### PR TITLE
Use SpecializedTasks where appropriate

### PR DESCRIPTION
**Pull request by sharwell**
* Update `SpecializedTasks.Default<T>` for C# 9
* Use `SpecializedTasks` where appropriate

### CODE REVIEW

File: src/Features/Core/Portable/Completion/CommonCompletionProvider.cs [hunk 1]
```diff
@@ -92,7 +93,7 @@ public override async Task<CompletionChange> GetChangeAsync(Document document, C
             => GetTextChangeAsync(selectedItem, ch, cancellationToken);
 
         protected virtual Task<TextChange?> GetTextChangeAsync(CompletionItem selectedItem, char? ch, CancellationToken cancellationToken)
-            => Task.FromResult<TextChange?>(null);
```
Comment (tmat) [inline src/Features/Core/Portable/Completion/CommonCompletionProvider.cs:95]: >Task.FromResult<TextChange?>(null); [](start = 15, length = 35)

Isn't null optimized in the Task class?

File: src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs [hunk 2]
```diff
@@ -365,7 +365,7 @@ public Task<bool> HasChangesAsync(Solution solution, string? sourceFilePath, Can
             var editSession = _editSession;
             if (editSession == null)
             {
-                return Task.FromResult(false);
```
Comment (tmat) [inline src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs:368]: >Task.FromResult(false); [](start = 23, length = 23)

I believe Boolean is optimized in Task.FromResult, we don't need ours.

File: src/Features/Core/Portable/Completion/CommonCompletionProvider.cs (hunk 1 above)
Comment (sharwell) [reply src/Features/Core/Portable/Completion/CommonCompletionProvider.cs:95]: No. `Task.FromResult<T>` always returns a new instance.

File: src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs (hunk 2 above)
Comment (sharwell) [reply src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs:368]: See https://github.com/dotnet/roslyn/pull/47117#discussion_r476756251. `Task.FromResult<TResult>` does not optimize for any cases.

Comment (sharwell) [reply src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs:368]: You may be thinking of `AsyncTaskMethodBuilder<TResult>`, which does optimize these cases for `async` methods that return synchronously.

Comment (tmat) [reply src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs:368]: I see. You are right

**Pull request merged**